src/edu/stanford/nlp/mt/tm/SampledRule.java

```diff
@@ -109,7 +112,7 @@ public int hashCode() {
    * 
    * @return
    */
-  public int[] e2fAll() {
+  public Set<Integer>[] e2fAll() {
```

Comment:
I'm confused.  Normally we're converting `int[][]` -> `Set<Integer>[]`, but here we're converting `int[]` -> `Set<Integer>[]`.  This seems like a dimension.  Is that correct?

Reply:
Using target->source alignments for explanation, the alignments are supposed to be 2 dimensional arrays, like int[len(tgt_tokens)][len(src_tokens)]. When the flatten function was used, the 2nd dimension was replaced by an integer. So, the new alignments were int[len(tgt_tokens)]. 

Since the alignments are never flattened, they would be in the Set<Integer>[] data structure.